contracts/operators/Beefy/BeefyZapBiswapLPVaultOperator.sol

```diff
@@ -0,0 +1,287 @@
+    }
+
+    /// @dev Zap one of the paired tokens for the LP Token, deposit the
+    ///         asset in the Beefy vault and receive the vault token (moo)
+    /// @param router The uniswap v2 router address to use for swaping and adding liquidity
+    /// @param vault The vault address to deposit into
+    /// @param token The token to zap
+    /// @param amount The token amount to deposit
+    function _zapAndStakeLp(
+        address router,
+        IBeefyVaultV6 vault,
+        address token,
+        uint256 amount
+    ) private {
+        IBiswapRouter02 biswapRouter = IBiswapRouter02(router);
+        IBiswapPair pair = IBiswapPair(vault.want());
+
+        require(pair.factory() == biswapRouter.factory(), "BLVO: INVALID_VAULT");
+
+        ExchangeHelpers.setMaxAllowance(IERC20(address(pair)), address(vault));
+
+        address cachedToken0 = pair.token0();
+        address cachedToken1 = pair.token1();
+
+        ExchangeHelpers.setMaxAllowance(IERC20(cachedToken0), router);
+        ExchangeHelpers.setMaxAllowance(IERC20(cachedToken1), router);
+
+        bool isInput0 = cachedToken0 == token;
+        require(isInput0 || cachedToken1 == token, "BLVO: INVALID_INPUT_TOKEN");
+
+        address[] memory path = new address[](2);
+        path[0] = token;
+
+        if (isInput0) {
+            path[1] = cachedToken1;
+        } else {
+            path[1] = cachedToken0;
+        }
+
+        (uint256 reserve0, uint256 reserve1, ) = pair.getReserves();
+
+        // The amount of input token to swap
+        // to get the same value of output token
+        uint256 swapAmountIn;
+        if (isInput0) {
+            swapAmountIn = _getOptimalSwapAmount(amount, reserve0, reserve1, biswapRouter, pair);
+        } else {
+            swapAmountIn = _getOptimalSwapAmount(amount, reserve1, reserve0, biswapRouter, pair);
+        }
+
+        uint256 lpAmount = _swapAndAddLiquidity(amount, swapAmountIn, path, biswapRouter);
+        vault.deposit(lpAmount);
+    }
+
+    /// @dev Swap input tokenA into TokenB to get the same value in tokenA
+    ///         as in tokenB to then add liquidity and store the obtained LP
+    ///         token in the vault beefy.
+    /// Note : path.length must be equal to 2 with path[0] = tokenA and path[1] = tokenB
+    /// @param amount The amount of tokenA to invest
+    /// @param swapAmountIn The amount of tokenA to swap for tokenB
+    /// @param path An array of the two paired token addresses
+    /// @param biswapRouter The uniswapV2 router to be used for swap and liquidity addition
+    function _swapAndAddLiquidity(
+        uint256 amount,
+        uint256 swapAmountIn,
+        address[] memory path,
+        IBiswapRouter02 biswapRouter
+    ) private returns (uint256 mintedLpAmount) {
+        uint256[] memory swappedAmounts = biswapRouter.swapExactTokensForTokens(
+            swapAmountIn,
+            1,
+            path,
+            address(this),
+            block.timestamp
```

Comment:
❔ Why is the interface expecting the block timestamp for a swap? Please enlighten me, I am not very familiar with the Biswap router

Reply:
You can specify the unix timestamp after which the transaction will revert (for e.g I want to execute my transaction and i need my tokens to be swapped in 1 min max).

Reply:
![image](https://user-images.githubusercontent.com/22816913/168114787-a354ef97-c790-46b1-a0f1-55d6159074f3.png)